ds3-autogen-java/src/main/java/com/spectralogic/ds3autogen/java/generators/requestmodels/BaseRequestGenerator.java

```diff
@@ -29,128 +32,106 @@
+        final String requestName = getRequestName(ds3Request.getName());
+        final String requestPath = getRequestPath(ds3Request);
+        final ImmutableList<Arguments> requiredArguments = toRequiredArgumentsList(ds3Request);
+        final ImmutableList<Arguments> optionalArguments = toOptionalArgumentsList(ds3Request.getOptionalQueryParams());
+        final ImmutableList<Arguments> constructorArguments = toConstructorArgumentsList(ds3Request);
+        final ImmutableList<Variable> classVariableArguments = toClassVariableArguments(ds3Request);
+        final ImmutableList<String> imports = getAllImports(ds3Request, packageName);
 
-    /**
-     * Converts data sorted within this RequestConvert into a Request model
-     * @return A Request model
-     */
-    private Request convert() {
-        final String[] classParts = ds3Request.getName().split("\\.");
-
-        return new Request(packageName,
-                classParts[classParts.length - 1],
+        return new Request(
+                packageName,
+                requestName,
                 ds3Request.getHttpVerb(),
-                requestPath(ds3Request),
+                requestPath,
                 ds3Request.getOperation(),
                 ds3Request.getAction(),
-                requiredConstructorArguments,
+                requiredArguments,
                 optionalArguments,
                 constructorArguments,
+                classVariableArguments,
                 imports);
     }
 
     /**
-     * Converts a Ds3Request and a package name into a Request model
+     * Gets all the class variables to properly generate the variables and their
+     * getter functions. This consists of all constructor arguments and optional
+     * arguments being converted into variables.
+     */
+    protected ImmutableList<Variable> toClassVariableArguments(final Ds3Request ds3Request) {
+        final ImmutableList.Builder<Variable> builder = ImmutableList.builder();
+        for (final Arguments arg : toConstructorArgumentsList(ds3Request)) {
+            builder.add(new Variable(arg.getName(), arg.getType(), true));
+        }
+        for (final Arguments arg : toOptionalArgumentsList(ds3Request.getOptionalQueryParams())) {
+            builder.add(new Variable(arg.getName(), arg.getType(), false));
+        }
+        return builder.build();
+    }
+
+    /**
+     * Gets all the required imports that the Request will need in order to properly
+     * generate the Java request code
      * @param ds3Request A Ds3Request
-     * @param packageName The name of the Java package for the generated request and response
-     * @return A Request model containing the information of the Ds3Request and package name
+     * @return The list of all imports that the Request requires for generating the
+     *         Java request code
      */
-    public static Request toRequest(
+    protected ImmutableList<String> getAllImports(
```

Comment:
Yes. All functions that are overridden are not static, but all those that are overridden are static.